src/main/webapp/script.js

```diff
@@ -220,51 +220,117 @@ function getGraphDisplay(graphNodes, graphEdges) {
     node.tip.show();
   });
 
-  const searchElement = document.getElementById('search');
-  document.getElementById('search-button').onclick = function() {
-    if (searchNode(cy, searchElement.value) || searchElement.value == "") {
-      document.getElementById('search-error').innerText = "";
+  document.getElementById('reset').onclick = function(){ resetElements(cy) };
+
+  document.getElementById('search-button').onclick = function() { search(cy, "node", searchNode) };
+
+  document.getElementById('search-token-button').onclick = function() { search(cy, "token", searchToken) };
+}
+
+/**
+ * Searches based on type (node or token)
+ */
+function search(cy, type, searchFunction) {
+  resetElements(cy);
+  let errorText = "";
+  let query = document.getElementById(type + '-search').value;
+  let result;
+  if (query != "") {
+    result = searchFunction(cy, query);
+    if (result) {
+      errorText = "";
     } else {
-      document.getElementById('search-error').innerText = "Node does not exist.";
+      errorText = type + " does not exist.";
     }
-  };
-  
+  }
+  document.getElementById(type + '-error').innerText = errorText;
+  return result;
 }
 
 /**
- * Zooms in on specific node
+ * Finds specific node and zooms in
  */
 function searchNode(cy, query) {
-  // reset nodes to default color
+  if (query) {
+    const target = cy.$id(query);
+    if (target.length != 0) {
+      highlightElements(cy, target);
+      cy.fit(target, 50);
+      return target;
+    }
+  }
+}
+/**
+ * Constructs list of nodes that contain specified token
+ * and zooms in
+ */
+function searchToken(cy, query) {
+  let target = cy.collection();
   cy.nodes().forEach(node => {
-    node.style('background-color', 'blue');
-    node.style('opacity', '1')
+    if (node.data().tokens.includes(query)) {
+      target = target.add(node);
+    }
   });
-  const target = findNodeInGraph(cy, query);
-  if (target) {
-    cy.nodes().forEach(node => node.style('opacity', '0.25'));
-    target.style('background-color', 'olive');
-    target.style('opacity', '1');
-    cy.fit(target, 50);
-    return true;
-  } else {
-    // fits all nodes on screen
-    cy.fit(cy.nodes(), 50);
-    return false;
+  if (target.length > 0) {
+    let showNode = target[0][0];
+    showNode.tip.show();
```

Comment:
design question: Should we show the token list when highlighting? This only shows that of the first node in the graph, which might block the view of other nodes and might not be the one the user actually want to see.
Maybe it is sufficient to only highlight the nodes (and not display token list) and if the user is interested in seeing the token list they can manually click on the node they are interested in?